integration-cli/docker_cli_service_logs_experimental_test.go

```diff
@@ -213,3 +213,69 @@ func (s *DockerSwarmSuite) TestServiceLogsFollow(c *check.C) {
 
 	c.Assert(cmd.Process.Kill(), checker.IsNil)
 }
+
+func (s *DockerSwarmSuite) TestServiceLogsTaskLogs(c *check.C) {
+	testRequires(c, ExperimentalDaemon)
+
+	d := s.AddDaemon(c, true, true)
+
+	name := "TestServicelogsTaskLogs"
+	replicas := 2
+
+	out, err := d.Cmd(
+		// create a service with the name
+		"service", "create", "--name", name,
+		// which has some number of replicas
+		fmt.Sprintf("--replicas=%v", replicas),
+		// which has this the task id as an environment variable templated in
+		"--env", "TASK={{.Task.ID}}",
+		// and runs this command to print exaclty 6 logs lines
+		"busybox", "sh", "-c", "for line in $(seq 1 6); do echo $TASK log test $line; done; sleep 100000",
+	)
+	c.Assert(err, checker.IsNil)
+	c.Assert(strings.TrimSpace(out), checker.Not(checker.Equals), "")
+
+	// make sure task has been deployed.
+	waitAndAssert(c, defaultReconciliationTimeout, d.CheckActiveContainerCount, checker.Equals, replicas)
+	waitAndAssert(c, defaultReconciliationTimeout, countLogLines(d, name), checker.Equals, 6*replicas)
+
+	// get the task ids
+	out, err = d.Cmd("service", "ps", "-q", name)
+	c.Assert(err, checker.IsNil)
+	// make sure we have two tasks
+	taskIDs := strings.Split(strings.TrimSpace(out), "\n")
+	c.Assert(taskIDs, checker.HasLen, replicas)
+
+	for _, taskID := range taskIDs {
+		c.Logf("checking task %v", taskID)
+		args := []string{"service", "logs", "--task", taskID}
+		cmd := exec.Command(dockerBinary, d.PrependHostArg(args)...)
+		r, w := io.Pipe()
+		cmd.Stdout = w
+		cmd.Stderr = w
+		c.Assert(cmd.Start(), checker.IsNil)
```

Comment:
Please use `github.com/docker/docker/pkg/testutil/cmd` to run commands in tests. It helps ensure that we have consistent (and sufficiently detailed) error messages when things fail, and it should be a lot less verbose.